### PR TITLE
Add support for macOS wheel

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ci/144)

### CODE REVIEW

**Review by BetsyMcPhail — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ci/144#-:-MzM5Sk55GrMcKtQGbQ9:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @mwoehlke-kitware)

___
*[driver/configurations/bazel.cmake, line 293 at r1](https://reviewable.io/reviews/robotlocomotion/drake-ci/144#-MzM5-mn0xm4Ofso7KD4:-MzM5-mo5OsywSPu1JDR:b-xtqnaj) ([raw file](https://github.com/robotlocomotion/drake-ci/blob/05bc62dfa45b4341892c4c805574dc0879bd0644/driver/configurations/bazel.cmake#L293)):*
> ```CMake
> if(PACKAGE)
>   set(DASHBOARD_PACKAGE_OUTPUT_DIRECTORY "/opt/drake")
>   mkdir("${DASHBOARD_PACKAGE_OUTPUT_DIRECTORY}" 1777
> ```

BTW Double-checking the switch from 0777 to 1777 was intentional

**Comment by mwoehlke-kitware** [comment]
___
*[driver/configurations/bazel.cmake, line 293 at r1](https://reviewable.io/reviews/robotlocomotion/drake-ci/144#-MzM5-mn0xm4Ofso7KD4:-MzM93bv5eUGoFIo186d:bc9ux4h) ([raw file](https://github.com/robotlocomotion/drake-ci/blob/05bc62dfa45b4341892c4c805574dc0879bd0644/driver/configurations/bazel.cmake#L293)):*
Yes; 0o1000 is the [sticky bit](https://en.wikipedia.org/wiki/Sticky_bit). Note `man 7 sticky`:

> A file in a sticky directory may only be removed or renamed by a user if the user has write permission for the directory and the user is the owner of the file, the owner of the directory, or the super-user.

In other words, it limits the cross-user permissions in an otherwise globally writable directory (both of us can add files, but I can't delete yours, and you can't delete mine). It *probably* doesn't matter for CI, but I consider it generally good practice, which is why I used it for the new wheel bits. Changing it here also helps consistency.

BTW, it would be good to run this on some Bazel build first, as well, as this (and `chmod`) also gained a `sudo`. (It looks like the wheel builds should already exercise the change to `chmod` via `aws.cmake, but a Bazel build should do so also.

**Comment by mwoehlke-kitware** [comment]
I should also add, I'm not sure there is great urgency to merge this, since it isn't usable until robotlocomotion/drake#16872 lands. (OTOH, having the CI changes "ready to go" may help land that, especially as it will allow requesting CI builds from that PR.)

**Review by BetsyMcPhail — APPROVED** [review]
This job: https://drake-jenkins.csail.mit.edu/view/Experimental/job/linux-bionic-gcc-bazel-experimental-release/14977/ was already set-up to run CI changes, so let's give it at try there

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ci/144#-:-MzMBUvkC95E_apFy77m:bld2e7z)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @mwoehlke-kitware)

**Comment by mwoehlke-kitware** [comment]
:+1:

**Comment by mwoehlke-kitware** [comment]
On second thought, I don't see that build fetching the AWS key. Maybe we need an unprovisioned build?

**Comment by mwoehlke-kitware** [comment]
Specifically, I would feel better if we see these two messages:
```
Downloading private key file from AWS S3...
```
```
Creating package output directory...
```

**Review by BetsyMcPhail — APPROVED** [review]
AFAICT CI defines the SSH_PRIVATE_KEY_FILE environment variable and never goes down the `Downloading private key file from AWS S3...` path. The `Creating package output directory...` would be a packaging job, I will launch one of those.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ci/144#-:-MzQjIK07b-58rdSbSvB:b-ipaolv)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @mwoehlke-kitware)

**Review by BetsyMcPhail — APPROVED** [review]
https://drake-jenkins.csail.mit.edu/view/Experimental/job/linux-focal-unprovisioned-gcc-bazel-experimental-snopt-packaging/31/

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ci/144#-:-MzRCQc5CinC_0sLPiqY:bharcb7)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @mwoehlke-kitware)

**Comment by mwoehlke-kitware** [comment]
:+1:
